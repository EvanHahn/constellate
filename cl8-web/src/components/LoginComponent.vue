<template>
<div>
  <div role="status" aria-live="polite">
    </div>
  <form v-on:submit.prevent="signIn">

    <div class="">
      <small v-if="errors && errors.has('email')"
        class="help is-danger">
        {{ errors.first('email') }}
      </small>
    </div>
    <input type="text" name="email" v-model="email"
      class="input-reset pa2 ba mb3 mt1 w-70"
      :class="{'bg-light-red': errors && errors.has('email') }"
      placeholder="your email address"
      aria-label="your email address"
      v-validate="'required|email'"
      autocomplete="email" >


    <input type="password" name="password" v-model="password"
      class="input-reset pa2 ba w-70"
      placeholder="your password"
      aria-label="your password" autocomplete="current-password">

    <div class="mt2">
      <button
        class="f6 link dim br-pill ph3 pv2 mb2 dib bg-light-silver white w-50 ml4 mt2"
        :class="{'bg-light-green': validity }"
        :disabled="errors && errors.any()"
        type="submit" name="button">
        Sign in
      </button>
    </div>

  </form>

</div>
</template>

<script>

export default {
  name: 'LoginComponent',
  props: ['fbase'],
  data: function () {
    return {
      email: '',
      password: null
    }
  },
  methods: {
    signIn: function () {
      // console.log(this.fbase)
      // let user = {
      //   email: this.email,
      //   password: this.password
      // }
      console.log(this.$validator.fields)
      // let loginResponse = this.fbase.handleAuthentication(user)
      // console.log(loginResponse)
    }
  },
  computed: {
    validity: function () {
      console.log(this)
    }
  }
}
</script>

<style>
@import '../../node_modules/tachyons/css/tachyons.css';

.sign-in-prompt{
  width: 500px;
  height: 500px;
  margin-left:auto;
  margin-right:auto;
  background-image: url(../assets/earth-transparent.png);
  background-repeat: no-repeat;
}
.sign-in-prompt a{

  top: 175px;
  left: 145px;
}


.bg-network{

}
</style>
