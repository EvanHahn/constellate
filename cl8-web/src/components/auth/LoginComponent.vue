<template>
<div class="w-100">

  <div v-if="loading" class="">
    <div class="spinner">
    <img src="../../assets/loading.svg" alt="loading"/>
  </div>
  </div>

  <div v-if="!loading" class="">

  <div class="v-mid sign-in-prompt">

    <h2 class="pt5 pl6">Constellation</h2>
    <div class="w-100 pl6">
      <div role="status" aria-live="polite" class="vh">
        <!--
      when there is an error, we want list it in here, to a screen reader
      can pick it up and read out the announcement
      -->
        <div v-if="errors" class="errors">
          <p v-for="(key, val) in errors.all()">
            {{ key }}
          </p>
        </div>
      </div>
      <form v-on:submit.prevent="signIn">

        <div class="w-100 mb3">

          <input type="text" name="email" v-model="email" class="input-reset pa2 ba  mt1"
            :class=" {'bg-washed-red': errors ? errors.has('email') : null}"
            placeholder="your email address"
            aria-label="your email address"
            v-validate="'required|email'"
            autocomplete="email"
          />

          <small v-if="errors && errors.has('email')" class="red ">
          {{ errors.first('email') }}
        </small>

        </div>

        <div class="w-100">
          <input type="password" name="password" v-model="password"
            class="input-reset pa2 ba" :class=" {'bg-washed-red': errors && errors.has('password') }"
            placeholder="your password"
            v-validate="'required|min:6'"
            aria-label="your password"
            autocomplete="current-password"
            @input="checkForValidFormSubmission" />

          <small v-if="errors && errors.has('password')" class="red ">
          {{ errors.first('password') }}
        </small>
        </div>

        <div class="mt2">
          <button class="f6 link br-pill ph3 pv2 mb2 bg-light-silver white w-50 ml0 mt2" :class="{'bg-light-green pointer grow': formValid}" :disabled="!formValid" v-bind:id="formValid" type="submit" name="button">
            Sign in
          </button>
        </div>

        <router-link :to="{ name: 'resetPassword' }">Forgot password</router-link>


      </form>

    </div>
  </div>
  </div>

</div>
</template>

<script>
/* eslint-disable */
import debugLib from 'debug'
const debug = debugLib('cl8.LoginComponent')
debug('sign in page')


export default {
  name: 'LoginComponent',
  data: function() {
    return {
      email: '',
      password: null,
      announcement: '',
      formIsValid: false
    }
  },
  methods: {
    signIn: function () {
      let user = {
        email: this.email,
        password: this.password
      }
      debug(user)
      this.$store.dispatch('login', user)
    },
    checkForValidFormSubmission: function() {
      let validation = {
        email: this.email,
        password: this.password
      }
      return this.$validator.validateAll(validation)
        .then(result => {
          if (!result) {
            this.formIsValid = result
            return false
          }
          debug(result)
          this.formIsValid = result
          return result
        })
        .catch(err => {
          debug(err)
        })
    }
  },
  computed: {
    formValid: function () {
      return this.formIsValid
    },
    loading: function () {
      return this.$store.getters.isLoading
    }
  }
}
</script>

<style>
@import '../../../node_modules/tachyons/css/tachyons.css';

.sign-in-prompt {
  width: 500px;
  height: 500px;
  margin-left: auto;
  margin-right: auto;
  background-image: url(../../assets/earth-transparent.png);
  background-repeat: no-repeat;
}

.sign-in-prompt a {

  top: 175px;
  left: 145px;
}

.vh {
  position: absolute !important;
  clip: rect(1px, 1px, 1px, 1px);
  padding: 0 !important;
  border: 0 !important;
  height: 1px !important;
  width: 1px !important;
  overflow: hidden;
}

.spinner {
  position: absolute;
  display: flex;
  justify-content: center;
  height: 100vh;
  width: 100vw;
  background-color: white;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  }
</style>
