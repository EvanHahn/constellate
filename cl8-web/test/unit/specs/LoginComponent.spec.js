
import { mount, shallow, createLocalVue } from 'vue-test-utils'
import VeeValidate from 'vee-validate'
import LoginComponent from '@/components/LoginComponent'



// we need to augment our Vue object for this component test
// as we're not evaluating any setup in main.js
const localVue = createLocalVue()
const config = { events: 'blur' }
localVue.use(VeeValidate, config)


// importing this file, because imports EventEmitter, which uses new syntax,
// giving an  an error alone the lines of `SyntaxError: Unexpected token export`

// how to I transpile it so jest doesn't choke on it?

// it turns out the solution is pass in a mock, and don't try to import es6-like
// that uses export statements

// TODO add tests   later, because fighting with babel will take hours to learn
// and reading the mocks docs and getting that will take hour s too
// import FirebaseService from '@/persistence/FirebaseService'


// check that a component renders

// check the interaction update the internal state of the component

// no need to check the rendering as Vue already has this covered

describe('Login Component', () => {

  // declare variable to overwrite
  let $mounted
  let otherMount

  describe('mounts with expected data', () => {
    let minimumData = { fbase: {} }

    beforeEach(() => {
      // $mount mounts the component, using the same lifecycle method
      // as vue itself uses
      // $mounted = new Vue(LoginComponent).$mount({propsData: minimumData})

      // whereas mount is the one defined in the test utils
      otherMount = mount(LoginComponent, {propsData: minimumData})
    })

    test('login mounts', () => {
      expect(otherMount.html()).toMatchSnapshot()
    })

    test('validSubmission', () => {

      // input when there's no data should be invalid
      expect().toBe(1)
    })
  })





  // the check that hitting when submit is triggered,
  // the onsubmit function is called
})
