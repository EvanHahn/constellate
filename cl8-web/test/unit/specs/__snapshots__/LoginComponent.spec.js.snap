// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component mounts with expected data login mounts 1`] = `
"<div><div role=\\"status\\" aria-live=\\"polite\\"></div> <form><input type=\\"text\\" name=\\"email\\" placeholder=\\"your email address\\" autocomplete=\\"email\\" aria-label=\\"your email address\\" class=\\"input-reset pa2 ba\\"> <!----> <input type=\\"password\\" name=\\"password\\" placeholder=\\"your password\\" aria-label=\\"your password\\" autocomplete=\\"current-password\\" class=\\"input-reset pa2 ba\\"> <button type=\\"submit\\" name=\\"button\\" class=\\"f6 link dim br-pill ph3 pv2 mb2 dib white bg-black\\">
      Sign in
    </button></form></div>"
`;
